File: third.rb
def count_even_digits(num)
  count = 0
  num = num.abs
  
  begin
    digit = num % 10
    count += 1 if digit % 2 == 0
    num /= 10
  end until num == 0
  
  count
end

print "Введите кол-во чисел: "
num = gets.chomp.to_i

(0...num).each do |i|
  print "Введите число #{i + 1}: "
  j = gets.chomp.to_i
  puts "кол-во четных цифр в числе #{j}: "
  puts count_even_digits(j)
end

File: third.cs
using System;

class Program
{
    static int CountEvenDigits(int num)
    {
        int count = 0;
        num = Math.Abs(num);
        
        do
        {
            int digit = num % 10;
            if (digit % 2 == 0) count++;
            num /= 10;
        } while (num != 0);
        
        return count;
    }

    static void Main()
    {
        Console.Write("Введите кол-во чисел: ");
        int num = int.Parse(Console.ReadLine());
        
        for (int i = 0; i < num; i++)
        {
            int j = int.Parse(Console.ReadLine());
            Console.WriteLine($"кол-во четных цифр в каждом числе: {j}: ");
            Console.WriteLine(CountEvenDigits(j));
        }
    }
}